Makefile

```diff
@@ -20,6 +20,13 @@ docker:
 	docker build --tag sail:0.1 .
 	@echo 'for example: docker run --volume `PWD`:/data/ sail:0.1 --help'
 
+install-deps:
+	@opam install . --deps-only
+
+check:
+	@echo "Running a pre-install check"
+	@ocaml -version | grep '4\.[1-9][3-9]' > /dev/null && echo "    Correct OCaml version found" || (echo "OCaml version >4.13 is required" && false)
```

Comment:
I'm not sure about this. It's usually advised to check for what you actually need, rather than a version of something. That being said, I wondered if there were already uses of `String.starts_with` in the sail repo. I didn't find any, but I did find local functions:
- In `src/lib/value.ml`:
  ```
  let value_string_startswith = function
  ```
- In `src/lib/sail_lib.ml`:
  ```
  let string_startswith (str1, str2) =
  ```

The latter function looks particularly interesting, and I'm wondering if we should port the current code to use that instead of `String.starts_with`, and avoid the worry about OCaml version.

Reply:
Makes sense, I can look into this!